src/lib.rs

```diff
@@ -33,12 +33,40 @@ assert_eq!(<X as Integer>::to_i32(), 7);
 ```
 
 Documented in each module is the full list of type operators implemented.
-*/
-#![cfg_attr(feature="no_std", no_std)]
-#[cfg(feature="no_std")]
-extern crate core as std;
-
-use std::cmp::Ordering;
+ */
+#![no_std]
+
+use core::cmp::Ordering;
+
+macro_rules! impl_derivable {
+    ($Type: ty) => (
+        impl ::core::cmp::PartialEq for $Type {
+            fn eq(&self, _: &Self) -> bool { match *self {} }
```

Comment:
Yeah, that's roughly what `#[derive]` would do. Also correct, as the match covers all (zero) cases.